src/curve_intersections.rs

```diff
@@ -0,0 +1,1377 @@
+        }
+    }
+
+    fn test_overlapping<T: ParamCurveBezierClipping>(curve: &T, t1: f64, t2: f64, accuracy: f64) {
+        assert!(t1 < t2);
+
+        do_double_test(
+            &curve.subsegment(0.0..t2),
+            &curve.subsegment(t1..1.0),
+            2,
+            CurveIntersectionFlags::KEEP_ALL_ENDPOINT_INTERSECTIONS, // Include endpoint intersections
+            accuracy,
+        );
+        do_double_test(
+            &curve.subsegment(0.0..t2),
+            &curve.subsegment(1.0..t1),
+            2,
+            CurveIntersectionFlags::KEEP_ALL_ENDPOINT_INTERSECTIONS, // Include endpoint intersections
+            accuracy,
+        );
+        do_double_test(
+            &curve.subsegment(0.0..1.0),
+            &curve.subsegment(t1..t2),
+            2,
+            CurveIntersectionFlags::KEEP_ALL_ENDPOINT_INTERSECTIONS, // Include endpoint intersections
+            accuracy,
+        );
+        do_double_test(
+            &curve.subsegment(0.0..1.0),
+            &curve.subsegment(t2..t1),
+            2,
+            CurveIntersectionFlags::KEEP_ALL_ENDPOINT_INTERSECTIONS, // Include endpoint intersections
+            accuracy,
+        );
+
+        do_single_test(
+            &curve.subsegment(0.0..1.0),
+            &curve.subsegment(t1..t2),
+            1,
+            CurveIntersectionFlags::KEEP_CURVE2_T0_INTERSECTION, // Only include curve2 t=0 endpoint intersection
+            accuracy,
+        );
+        do_single_test(
+            &curve.subsegment(t1..t2),
+            &curve.subsegment(0.0..1.0),
+            0,
+            CurveIntersectionFlags::KEEP_CURVE2_T0_INTERSECTION, // Only include curve2 t=0 endpoint intersection
+            accuracy,
+        );
+        do_single_test(
+            &curve.subsegment(0.0..1.0),
+            &curve.subsegment(t1..t2),
+            1,
+            CurveIntersectionFlags::KEEP_CURVE2_T1_INTERSECTION, // Only include curve2 t=1 endpoint intersection
+            accuracy,
+        );
+        do_single_test(
+            &curve.subsegment(t1..t2),
+            &curve.subsegment(0.0..1.0),
+            0,
+            CurveIntersectionFlags::KEEP_CURVE2_T1_INTERSECTION, // Only include curve2 t=1 endpoint intersection
+            accuracy,
+        );
+    }
+
+    #[test]
+    fn test_cubic_cubic_intersections() {
+        test_t(&CubicBez::new(
+            (0.0, 0.0),
+            (0.3, -1.0),
+            (0.7, -1.0),
+            (1.0, 0.0),
+        ));
+        test_t(&CubicBez::new(
+            (0.0, 0.0),
+            (-1.0, 0.0),
+            (1.0, 0.0),
+            (1.0, 0.0),
+        ));
+
+        do_double_test(
+            &CubicBez::new((0.0, 0.0), (0.0, 1.0), (0.0, 1.0), (1.0, 1.0)),
+            &CubicBez::new((0.0, 1.0), (1.0, 1.0), (1.0, 1.0), (1.0, 0.0)),
+            1,
+            CurveIntersectionFlags::NONE, // Standard algorithm
+            0.,                           // Tightest possible accuracy
```

Comment:
This is where that epsilon function comes in that you commented on before. The curve intersection function automatically uses the minimum recommended epsilon because 0 accuracy can never be achieved. However, I can see how this could be concerning from a numerical stance since the "minimum recommended epsilon" is fairly arbitrary. As such, what do you think would be the appropriate route to take? Maybe we should have a function called `suggested_min_epsilon` for floats, points, curves, and lines that the developer would have to call if they wanted the tightest accuracy?